### ISSUE TITLE
Investigate failing e2e tests for VPP 20.01

### ISSUE THREAD

**Issue by ondrej-fabry** [ligato/vpp-agent#1465]
Some e2e tests are failing for VPP 20.01 on Travis.

First failing build on Travis here: https://travis-ci.com/ligato/vpp-agent/builds/126617460

Failing test cases:
- [ ] [`--- FAIL: TestLinkOnly (13.13s)`](https://travis-ci.com/ligato/vpp-agent/jobs/232707974#L3293)
- [ ] [`--- FAIL: TestIPWithNeighGW (17.24s)`](https://travis-ci.com/ligato/vpp-agent/jobs/232707974#L3650)
- [ ] [`--- FAIL: TestIPWithNonLocalGW (17.24s)`](https://travis-ci.com/ligato/vpp-agent/jobs/232707974#L4016)
- [ ] [`--- FAIL: TestVPPRoutesWithNetalloc (18.25s)`](https://travis-ci.com/ligato/vpp-agent/jobs/232707974#L4341)

**Comment by rewenset** [comment]
seems like the issue is with `flags`
for example, to fix `TestIPWithNeighGW` you should change converting function, by adding `3` to it.
```go
# file: plugins/vpp/ifplugin/vppcalls/vpp2001/dump_interface_vppcalls.go
func ifAPIFlagState(flags vpp_ifs.IfStatusFlags) uint32 {
	if flags == vpp_ifs.IF_STATUS_API_FLAG_ADMIN_UP {
		return 1
	} else if flags == vpp_ifs.IF_STATUS_API_FLAG_LINK_UP {
		return 2
	} else if flags == 3 {
		return 3
	}
	return 0
}
```
Why `3`? Because it is what returned when interface is `enabled`.
```
loop0: 3 
```
But if `enabled = false` then `IF_STATUS_API_FLAG_LINK_UP` returned.
```
tap0: IF_STATUS_API_FLAG_LINK_UP 
```
And this is the reason, I think, behind `DownstreamResync` contains not only NOOP, but UPDATE too. With that `tap0` example:
```
+======================================================================================================================+
| #3 - NB Transaction                                                                                 DownstreamResync |
+======================================================================================================================+
  * transaction arguments:
      - seqNum: 3
      - type: NB Transaction, SB Sync

o----------------------------------------------------------------------------------------------------------------------o
  * executed operations (2019-09-25 12:08:02.442 +0000 UTC -> 2019-09-25 12:08:02.456 +0000 UTC, dur: 13ms):
      1. UPDATE:
          - key: config/vpp/v2/interfaces/vpp-tap
          - prev-value: { name:"vpp-tap" type:TAP enabled:true phys_address:"02:fe:5f:ad:a4:9d" tap:<version:2 >  }
          - new-value: { name:"vpp-tap" type:TAP tap:<version:2 >  }
x----------------------------------------------------------------------------------------------------------------------x
| #3 - NB Transaction                                                                                      took 13.3ms |
x----------------------------------------------------------------------------------------------------------------------x
```

I'm still in progress of solving this issue. I will appreciate any help

**Comment by milanlenco** [comment]
Good catch!

> Why 3? Because it is what returned when interface is enabled.

Could you please also find the VPP commit that introduced this change and paste the commit message here? It is preferred to be aware of and understand the ongoing changes in VPP, rather than keep updating vppcalls solely based on observation. Then instead of `flags == 3` we could also define appropriate constant for the value and also for other possible values if there are more.
Thanks

**Comment by rewenset** [comment]
@milanlenco thanks :blush:
I want that too! But don't know where can I find sources. For now I'm only learning by testing, but reading sources will get my life a lot easier. I've already asked about it in PR, so no need to reply twice :wink:

**Comment by milanlenco** [comment]
20.01 is the upcoming version, so you can clone the gerrit/github VPP repo and checkout the master branch.

**Comment by ondrej-fabry** [comment]
Fixed in #1479
